### PR TITLE
Work on CancelDonation UC

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: src/UseCases/CancelDonation/CancelDonationUseCase.php [hunk 1]
```diff
@@ -22,12 +23,30 @@ public function __construct( DonationRepository $donationRepository, TemplateBas
 	}
 
 	public function cancelDonation( CancelDonationRequest $cancellationRequest ): CancelDonationResponse {
+		try {
+			$donation = $this->donationRepository->getDonationById( $cancellationRequest->getDonationId() );
+		}
+		catch ( GetDonationException $ex ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), false );
```
Comment (KaiNissen) [inline src/UseCases/CancelDonation/CancelDonationUseCase.php]: maybe use class constants for the boolean parameter?

File: src/UseCases/CancelDonation/CancelDonationUseCase.php [hunk 2]
```diff
@@ -22,12 +23,30 @@ public function __construct( DonationRepository $donationRepository, TemplateBas
...
+
+		if ( $donation === null ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), false );
+		}
+
+		try {
+			$donation->cancel();
+		}
+		catch ( \RuntimeException $ex ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), true );
```
Comment (KaiNissen) [inline src/UseCases/CancelDonation/CancelDonationUseCase.php]: this should be an unsuccessful cancellation too, right?

File: src/UseCases/CancelDonation/CancelDonationUseCase.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/UseCases/CancelDonation/CancelDonationUseCase.php]: It's still unclear to me what the response model should be (https://phabricator.wikimedia.org/T123046), so I'm not spending time on making this nice while not knowing if it's the right stuff to begin with

File: src/UseCases/CancelDonation/CancelDonationUseCase.php (hunk 2 above)
Comment (JeroenDeDauw) [inline src/UseCases/CancelDonation/CancelDonationUseCase.php]: Seems like someone wrote production code that was not needed to make the test pass...

**Pull request merged**